stackrox/stackrox
# ROX-13326: Use UUID columns for roles, permissionsets and simpleaccessscopes (PermissionSet and SimpleAccessScope identifiers) in postgres mode.

## Description

The relation between roles and the associated permission sets as well as access scopes used to be done relying on decorated UUIDs (generated UUIDs prefixed with human-readable strings).
These object are candidate to conversion to UUIDs.

Some of the UUID base work is needed for this change to work (PRs https://github.com/stackrox/stackrox/pull/3679 and https://github.com/stackrox/stackrox/pull/3681 ).

PR stack:
https://github.com/stackrox/stackrox/pull/3802 (server support code)
+- https://github.com/stackrox/stackrox/pull/3725 (data migration)
... +- https://github.com/stackrox/stackrox/pull/3818 (UI code)

## Checklist
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI is a good first step.

Manual testing performed:
- deploy central in non-postgres mode.
- check the existing roles, permission sets and access scopes.
- create a sample permission set, a sample access scopes, and roles covering the possible combinations of default and created permission sets and access scopes.
- open the existing roles, permission sets and access scopes and keep trace of their identifiers.
- deploy postgres central-db.
- switch central to postgres mode.
- monitor for migration steps. These should include migration 148 to 149, 157 to 158 and 163 to 164 (currently latest rocks migration sequence number is 111, and the altered migrations are postgres 37 to 38, 46 to 47 and 52 to 53).
- open the UI
- check the new identifiers. They should all be UUIDs. The identifiers that were prefixed UUIDs before the migration should be the same UUID without prefix.
- check that the inter-object relations are kept (roles referencing permission sets and access scopes as well as referenced permission sets and access scopes).
- connect to the postgres DB
- check the table datamodel (id columns for permission_sets and simple_access_scopes should be UUIDs).

One observation during the manual test was that the `Unrestricted` access scope page failed to open. A pull request has been opened in order to fix the UI part as a consequence (https://github.com/stackrox/stackrox/pull/3818).
